src/Microsoft.DotNet.Build.Tasks/PackageFiles/PerfTesting.targets

```diff
@@ -4,57 +4,94 @@
+    <CliExitErrorCommand>EXIT /B 1</CliExitErrorCommand>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetOS)'=='Linux'">
+    <PythonCommand>python3.5</PythonCommand>
+    <CliExitErrorCommand>exit 1</CliExitErrorCommand>
+  </PropertyGroup>
+
+  <!-- TODO: Attempt to fail if PerformanceType is unknnown. -->
+  <PropertyGroup>
+    <PerfTestCommand>$(PerfTestCommandDotnetExecutable) PerfRunner.exe --perf:runid $(RunId) --perf:collect $(CollectFlags) || $(CliExitErrorCommand)</PerfTestCommand>
+    <MeasurementPyCommand>$(PythonCommand) $([System.IO.Path]::Combine($(BenchviewDir), "measurement.py")) xunit "$(RunId)-$(AssemblyName).xml" --better desc --drop-first-value --append -o "$(ProjectDir)measurement.json"</MeasurementPyCommand>
+  </PropertyGroup>
+
   <ItemGroup>
     <PerfTestCommandLines Include="$(PerfTestCommand)" />
   </ItemGroup>
+
   <ItemGroup Condition="'$(TargetOS)'=='Windows_NT' and '$(LogToBenchview)' == 'true'">
-    <PerfTestCommandLines Include="if exist Perf-$(AssemblyName).xml (" />
-    <PerfTestCommandLines Include="py $(BenchviewDir)\measurement.py xunit Perf-$(AssemblyName).xml --better desc --drop-first-value --append -o $(ProjectDir)measurement.json" />
+    <PerfTestCommandLines Include="if exist &quot;$(RunId)-$(AssemblyName).xml&quot; (" />
+    <PerfTestCommandLines Include="$(MeasurementPyCommand) || $(CliExitErrorCommand)" />
     <PerfTestCommandLines Include=")" />
   </ItemGroup>
+
   <ItemGroup Condition="'$(TargetOS)'=='Linux' and '$(LogToBenchview)' == 'true'">
-    <PerfTestCommandLines Include="if [ -a Perf-$(AssemblyName).xml ]" />
-    <PerfTestCommandLines Include="then" />
-    <PerfTestCommandLines Include="python3.5 $(BenchviewDir)\measurement.py xunit Perf-$(AssemblyName).xml --better desc --drop-first-value --append -o $(ProjectDir)measurement.json" />
+    <PerfTestCommandLines Include="if [ -a &quot;$(RunId)-$(AssemblyName).xml&quot; ]; then" />
+    <PerfTestCommandLines Include="$(MeasurementPyCommand) || $(CliExitErrorCommand)" />
     <PerfTestCommandLines Include="fi" />
   </ItemGroup>
+
   <!-- Optimizations to configure Xunit for performance -->
   <ItemGroup Condition="'$(IncludePerformanceTests)' == 'true'">
     <AssemblyInfoUsings Include="using Microsoft.Xunit.Performance%3B" />
     <AssemblyInfoLines Include="[assembly:OptimizeForBenchmarks]" />
   </ItemGroup>
 
+  <PropertyGroup>
+    <SubmissionPyCommand>$(PythonCommand) $([System.IO.Path]::Combine($(BenchviewDir), "submission.py")) "$(ProjectDir)measurement.json" --build "$(ProjectDir)build.json" --machine-data "$(ProjectDir)machinedata.json" --metadata "$(ProjectDir)submission-metadata.json" --group "CoreFx" --type "$(BenchviewRuntype)" --config-name "$(ConfigurationGroup)" --config Configuration "$(ConfigurationGroup)" --config OS "$(TargetOS)" --config "RunType" "$(PerformanceType)" -arch "$(Platform)" --machinepool "PerfSnake" -o "$(ProjectDir)submission.json"</SubmissionPyCommand>
```

Comment:
Any reason you are using Path.Combine here? Seems like an easy concat like everywhere else would be appropriate and likely more efficient.

Reply:
In Windows is: `$(BenchviewDir)\\"submission.py"`, and on Linux: `$(BenchviewDir)/"submission.py"`. This way seemed cleaner and correct. Is there a huge performance penalty by using the function?

Reply:
@jorive I would expect it to be negligible, it is just readability.  I don't think it matters either way.

Reply:
note : python subdir/subdir2/foo.py works just fine in Windows.